src/llama_recipes/utils/train_utils.py

```diff
@@ -6,21 +6,49 @@
 import yaml
 from pathlib import Path
 from pkg_resources import packaging
-
+import contextlib
+import gc
 
 import torch
 import torch.cuda.nccl as nccl
 import torch.distributed as dist
 from torch.distributed.fsdp import StateDictType
 from torch.distributed.fsdp.sharded_grad_scaler import ShardedGradScaler
+# from torch.utils.flop_counter import FlopCounterMode
 from tqdm import tqdm
 from transformers import LlamaTokenizer
 
 
 from llama_recipes.model_checkpointing import save_model_checkpoint, save_model_and_optimizer_sharded, save_optimizer_checkpoint
 from llama_recipes.policies import fpSixteen,bfSixteen_mixed, get_llama_wrapper
 from llama_recipes.utils.memory_utils import MemoryTrace
-
+from llama_recipes.utils.tflop_counter import FlopCounterMode
+
+@contextlib.contextmanager
+def maybe_run_profiler(cfg, *args, **kwargs):
+    use_profiler: bool = cfg.profiler
+
+    if use_profiler:
+        with torch.profiler.profile(
```

Comment:
future idea - I usually like to have a log or print to let the user know that profiling is active, and where traces will be saved to.

Reply:
Good idea let me add it.